### ISSUE TITLE
Game feedback

### ISSUE THREAD

**Issue by Srogozins** [YeOldeDM/ultracool#1]
Creating this issue for bugs and other feedback. Figured out that keeping it all together could be more suitable for a small one-dev project.

**Gameplay**
I like the new hop-swap ability, it has quickly proved itself useful for escaping sticky situations as well as fooling the mooks into killing each other. That said, first time I tried it, when I wasn't fully sure what it does, I actually didn't notice it. Maybe add some sound and/or visual effect to make it more explicit. 
By the way, I assume you're going to implement scoring at some point, I'd suggest giving bonus score for harder kills, like, as mentioned, when you have mooks kill each other or even themselves.

**Crashes**
- Only got this one once. Prints following in debugger:
  `Attempt to call function 'is_queued_for_deletion' in base 'previously freed instance' on a null instance.`
  Stack frame(didn't find a way to export them so sorry if mistyped something):

```
0 - res://gd/bullet.gd:22
members
  owner 0
  splode 617
locals
  body 0
  s 0
```
- This keeps on happening, sometimes just after a few minutes, but usually it takes for it longer that 5 minutes to occur. Not 100% sure, but might be happening right after someone is killed. Following message is output to debugger: `Invalid set index 'owner' (on base: 'previously freed instance').` 

```
0 - res://scn/world.gd:97

members
  game 0
  player_start 50,42
  env 0
  soundtrack sord
  nav 0
  hopswap_label 0
 respawn_label 0
 xhair 0
 player 0
  hop_target
  mpos 17.25,22.75
  old_mpos 17.25,22.75
  DRAW_PATHS off
  music 0
  timescale off
  mooks Array[1]
  spawners Array[3]
  sounds Array[6]
locals
  mook 0
```

```
1 - res://scn/mook.gd:92

members
  path Vector2[3]
  kill 0
  shoot 0
  speed 14
  spawner 0
  my_buller 0
  world 0
  has_fov on
  can_shoot off
  draw_path on
  gib 608
  shoot_timer 1.851
  bullet 613
  sprite 0
  dead on
  shoot_time 3
locals
  lv 2.926293,4.77
```

```
1 - res://scn/mook.gd:34

members
  path Vector2[3]
  kill 0
  shoot 0
  speed 14
  spawner 0
  my_buller 0
  world 0
  has_fov on
  can_shoot off
  draw_path on
  gib 608
  shoot_timer 1.851
  bullet 613
  sprite 0
  dead on
  shoot_time 3
locals
  delta 0.017
```

Additional weirdness(not sure if specific to the game, but just in case): when clicking a stack frame, I'll sometimes get this in stderr:

```
WARNING: call: Can't call non-static function: 'get_path' in script.
   At: modules/gdscript/gd_script.cpp:1937.
SCRIPT ERROR: get_path: Cannot access member without instance.
   At: res://scn/mook.gd:95.
Stack frames:
```

**Gameplay bugs**
- If a mook kills you while you're aligned to hop-swap with him, the prompt "Press E to hop-swap" is still shown and pressing it teleports the mook to the position where you died.
- After a mooks are killed(changes color to dark red), they still follow the player for some time(even turning corners), before exploding into gibs.

**Audio issues:**
- ~~After prolonged gameplay(3-7 minutes) the music(background noise that changes in pitch depending on time flow) stops playing. I haven't noticed any specific patterns for reproducing it, except that it seems to happen after a mook is killed. The rest of in-game audio seems to continue playing normally.~~ Music removed now. Hope it can come back at some point.

**UI Annoyances:**
- ~~At the splash screen ("BathWater"), left-clicking to proceed to the main menu seems to send the click to it as well. For example, if you are on the splash screen and click while your mouse is right over where the start button is going to appear, the main menu flashes for a brief moment and the game proper is immediately started. Seems like the buttons are activated upon mouse button release, so it can be avoided by holding down the mouse button and releasing it after moving it away from the button.~~ Duct-tape fixed.

**Minor issues:**
- ~~Pressing `1` or `2`, in addition to the sound effect prints on or more instances error message below to console. It seems that if you hold down the key instead of very quickly tapping it(faster than I can), it keeps triggering the same input event, because the sound effect stutters and the amount error messages is directly proportional to how long the button was pressed. Error:~~ Ultra kewl sound effect actions removed.

```
ERROR: update: Invalid mix rate for voice (0) check for invalid pitch_scale param.
   At: servers/spatial_sound_2d/spatial_sound_2d_server_sw.cpp:1003.
```
- Upon starting the game (play the project), following error is printed to console: 

```
ERROR: update: Invalid mix rate for voice (0) check for invalid pitch_scale param.
   At: servers/spatial_sound_2d/spatial_sound_2d_server_sw.cpp:1003.
```
- Upon clicking `START` in the main menu, following error is printed to console:

```
ERROR: get_node: Node not found: env
   At: scene/main/node.cpp:929.
```

**Comment by YeOldeDM** [comment]
Report of bug fixes:
- scoring is in, and is displayed  
- fixed hop-swapping while being dead  
- music cut-out: is caused by the sample not being able to loop (for some reason). Music is out for now.  Fudge it.
- UI input bleed-over: now when entering the title scene, the mouse is warped to the topleft corner, far away from anything click-able. Duct-tape fix to be fixed better soon.
- "ultra" "kewl" samples bound to 1 2 have been disabled (they were a test toy that needs to be removed anyway)  
- Random errors in console: I've noticed those too. But they seem to have no impact on how the game works, so I am biting the bullet and not worrying about them. Hopefully that doesn't bite me in the ass later.

**Comment by YeOldeDM** [comment]
as a side note: I also managed to find+fix(??) a nasty little game-crashing bug which involved mook gibs being added into the same container as bullets.

**Comment by Srogozins** [comment]
Results from testing today(commit e5476e138f764f04d7e0ad114eebb15ab6646ce3)
- Still can hop-swap while dead.
- Started getting some random crashes(details in first post).
- Possible gameplay bug with dead mooks following the player.
- New sounds seem to be working fine.

**Comment by YeOldeDM** [comment]
_Still can hop-swap while dead._  I swear I fixed this. Oh well.
_Started getting some random crashes(details in first post)._  I over-hauled the system which handles sound effects, so things should be much more stable now. Not 100% solid, but good enough.
_Possible gameplay bug with dead mooks following the player._  This is sort-of intended (mooks retain some of their velocity after they die, before they gib.)
_New sounds seem to be working fine._ I finally got the god-damn friggin' music to loop ;)
